esmvaltool/preprocessor/__init__.py

```diff
@@ -384,12 +391,22 @@ def _run(self, _):
             logger.debug("Running block %s", block)
             if block[0] in MULTI_MODEL_FUNCTIONS:
                 for step in block:
+                    logger.debug("Applying step: %s", step)
                     self.products = _apply_multimodel(self.products, step,
                                                       self.debug)
             else:
                 for product in self.products:
                     logger.debug("Applying single-model steps to %s", product)
                     for step in block:
+                        logger.debug("**************************************")
+                        logger.debug("Applying step: %s", step)
+                        ref_product = product.settings['fix_file']
+                        logger.debug("Dataset: %s", ref_product['dataset'])
+                        logger.debug("Variable: %s", ref_product['short_name'])
+                        diag_name = \
+                            ref_product['output_dir'].split(os.sep)[-3]
```

Comment:
This feels a lot like a hack to me. It splits the output file path and prints part of it. If the output path changes in the future for whatever reason, it won't print the correct part and it would be very hard to figure out why.

The correct way would be to pass the information like the diagnostic name and preprocessor list name from the recipe. 

Either way, this method can't be used to get the name of the preprocessor list as it doesn't appear in the output path.

Reply:
maybe @bouweandela has a better idea? changing function positional arguments in some distant module just so we get the debug messages a bit clearer seems like a bit of a drag to me, but then again, I am not an advanced user :grin: